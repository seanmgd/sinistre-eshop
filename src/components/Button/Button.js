import PropTypes from 'prop-types'
import styled from 'styled-components'
import theme from '../../styles/theme'

const PADDINGS = { small: '4px', medium: '8px', large: '16px' }
const SIZES = Object.keys(PADDINGS)
const COLORS = Object.keys(theme.colors)

export const Button = styled.button`
  color: ${({ color, theme }) => theme.colors[color].light};
  padding: ${({ size }) => PADDINGS[size]};
  background-color: ${({ color, theme }) => theme.colors[color].base};
  ${({ disabled }) => disabled && `opacity: .5`};
  font-size: 14px;
  border-radius: 4px;
  font-weight: bold;
  text-align: center;
`

Button.propTypes = {
  color: PropTypes.oneOf(COLORS).isRequired,
  size: PropTypes.oneOf(SIZES).isRequired,
  disabled: PropTypes.bool,
}

Button.defaultProps = {
  color: 'primary',
  disabled: false,
  size: 'medium',
}

Button.COLORS = COLORS
Button.SIZES = SIZES
