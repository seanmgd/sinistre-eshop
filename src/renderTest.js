import { render as renderTest } from '@testing-library/react'
import { AppProviders } from './AppProviders'
import theme from './styles/theme'

// Export it first to allow overrides.
export * from '@testing-library/react'
export { renderInAppProviders as render }

export function renderInAppProviders(node, options) {
  const { rerender, ...res } = renderTest(
    <AppProviders theme={theme} children={node} />,
    options,
  )

  return {
    ...res,
    rerender: node => rerender(<AppProviders theme={theme} children={node} />),
  }
}
