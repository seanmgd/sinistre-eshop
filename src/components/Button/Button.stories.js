import { boolean, select, text } from '@storybook/addon-knobs'
import { Button } from './Button'

export default {
  title: 'Action/Button',
  component: Button,
}

export const ToStorybook = () => (
  <Button
    children={text('children', 'My button')}
    color={select('color', Button.COLORS, Button.defaultProps.color)}
    disabled={boolean('disabled', Button.defaultProps.disabled)}
    size={select('size', Button.SIZES, Button.defaultProps.size)}
  />
)

ToStorybook.story = {
  name: 'Button',
}
