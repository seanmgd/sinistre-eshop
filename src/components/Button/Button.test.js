import React from 'react'
import { render } from '../../renderTest'
import { Button } from './Button'

function createProps(props) {
  return { children: 'My button', ...props }
}

describe('<Button />', () => {
  it('should render a button', () => {
    const props = createProps()
    const { container } = render(<Button {...props} />)

    expect(container.firstChild.disabled).toBe(false)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a disabled button', () => {
    const props = createProps()
    const { container } = render(<Button {...props} disabled />)

    expect(container.firstChild.disabled).toBe(true)
  })

  Button.COLORS.forEach(color => {
    it(`should render a button with ${color} color`, () => {
      const props = createProps()
      const { container } = render(<Button {...props} color={color} />)

      expect(container.firstChild).toMatchSnapshot()
    })
  })

  Button.SIZES.forEach(size => {
    it(`should render a ${size} button`, () => {
      const props = createProps()
      const { container } = render(<Button {...props} size={size} />)

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
