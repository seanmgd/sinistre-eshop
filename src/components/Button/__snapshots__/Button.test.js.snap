// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button 1`] = `
<button
  class="sc-bdfBwQ iPyZpa"
  color="primary"
>
  My button
</button>
`;

exports[`<Button /> should render a button with black color 1`] = `
<button
  class="sc-bdfBwQ cwpJDj"
  color="black"
>
  My button
</button>
`;

exports[`<Button /> should render a button with error color 1`] = `
<button
  class="sc-bdfBwQ fYLblI"
  color="error"
>
  My button
</button>
`;

exports[`<Button /> should render a button with grey color 1`] = `
<button
  class="sc-bdfBwQ kEshBj"
  color="grey"
>
  My button
</button>
`;

exports[`<Button /> should render a button with primary color 1`] = `
<button
  class="sc-bdfBwQ iPyZpa"
  color="primary"
>
  My button
</button>
`;

exports[`<Button /> should render a large button 1`] = `
<button
  class="sc-bdfBwQ hLjxpl"
  color="primary"
>
  My button
</button>
`;

exports[`<Button /> should render a medium button 1`] = `
<button
  class="sc-bdfBwQ iPyZpa"
  color="primary"
>
  My button
</button>
`;

exports[`<Button /> should render a small button 1`] = `
<button
  class="sc-bdfBwQ kHCIhK"
  color="primary"
>
  My button
</button>
`;

exports[`<Button /> should render a x-large button 1`] = `
<button
  class="sc-bdfBwQ hlKBmJ"
  color="primary"
>
  My button
</button>
`;
